backend/src/main/java/com/team28/sidedish/controller/oauth/OAuthUrl.java

```diff
@@ -0,0 +1,64 @@
+package com.team28.sidedish.controller.oauth;
+
+import com.team28.sidedish.exception.URISyntaxException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OAuthUrl {
+
+    public static OAuthUrlBuilder builder() {
+        return new OAuthUrlBuilder();
+    }
```

Comment:
빌더 패턴을 잘 사용해 주셨네요 👍 

한가지 리뷰를 드리자면 빌더를 이용해서 url 생성에 필요한 정보를 가진 인스턴스를 생성하고, 
해당 인스턴스의 메소드를 사용해서 url을 반환받는 방식으로 개선하면 어떨까요?
```suggestion
public class GithubOAuth {
    private static final String GITHUB_OAUTH_LOGIN_URL = "https://github.com/login/oauth/authorize";
    private static final String GITHUB_OAUTH_SCOPE = "read:user";
    private static final String GITHUB_OAUTH_RESPONSE_TYPE = "code";
    
    private String baseUrl;
    private String clientId;
    // ...
    
    public String url(){
      return baseUrl + getQueryString();
    }
    
    // ...
    public static OAuthUrlBuilder builder() {
        return new OAuthUrlBuilder();
    }
    
    // ...
```

Reply:
단순히 String 을 반환하는 빌더보다는, URL 을 갖는 의미있는 객체로 만드는게 더 좋을 것 같네요!
감사합니다👍